homeassistant/helpers/entity_registry.py

```diff
@@ -1217,9 +1250,8 @@ def async_entries_for_device(
     """Return entries that match a device."""
     return [
         entry
-        for entry in registry.entities.values()
-        if entry.device_id == device_id
-        and (not entry.disabled_by or include_disabled_entities)
+        for entry in registry.entities.get_entries_for_device_id(device_id)
+        if (not entry.disabled_by or include_disabled_entities)
     ]
 
```

Comment:
async_entries_for_area isn't used very many places, where-as the other two are all over the place

Reply:
`dr.async_entries_for_config_entry` is called all over the place but its pretty easy to see why it doesn't show up very high on the profile by looking at only the file sizes:

```
homeassistant:/config/.storage# ls -l core.device_registry 
-rw-r--r--    1 root     root        338289 Jan  7 19:07 core.device_registry
homeassistant:/config/.storage# ls -l core.entity_registry 
-rw-r--r--    1 root     root       4215766 Jan  7 19:07 core.entity_registry
```

Reply:
Comment didn't age well about not needing one for areas.